### PR TITLE
Timeout RTD module: initial release

### PR DESCRIPTION
**Pull request by luigi-sayson**
## Type of change

- [x] Feature

## Description of change

Adding a new RTD module to manage timeouts in real time. The current implementation supports adding time based on rules that look at specific features (number of ad units in the request, device type, connection speed, and whether or not a video ad unit is present). These rules can either be retrieved from a timeout provider, or configured statically on page. 

## Other information
Please see related proposal issue: https://github.com/prebid/Prebid.js/issues/7393

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@luigi-sayson looks like there are a few linting errors in this pr. Can you please fix?

**Comment by luigi-sayson** [comment]
@ChrisHuie, done. Sorry, I ran `gulp test-coverage` last and didn't realize that it wasn't running lint

**Comment by patmmccann** [comment]
Connection speed slow/medium/fast almost always resolves to fast or missing. We see 4g for effective type on north of 90% of traffic with it filled in. This isn't supported in Safari, so I am curious to see added documentation on specifying the timeout on missing.

**Comment by luigi-sayson** [comment]
The current implementation should already handle a rule for 'unknown' connection speeds. I just forgot to add it to the documentation, which I've done now. 

> We see 4g for effective type on north of 90% of traffic with it filled in. 

Yeah the idea behind the connection speed is that most pubs should choose 'fast', but maybe some operate in a region with slower connection speeds and would want to adjust their timeouts accordingly

File: modules/timeoutRtdProvider.md
Comment (patmmccann) [inline modules/timeoutRtdProvider.md:15]: can you make it clear here that the contents of the response can also be set via config and no endpoint is required?

File: modules/timeoutRtdProvider.md
Comment (patmmccann) [inline modules/timeoutRtdProvider.md:79]: 'The rules retrieved from the endpoint or set via setConfig'

File: modules/timeoutRtdProvider.md
Comment (patmmccann) [inline modules/timeoutRtdProvider.md:100]: Can you add some detail about if there are mulitple rule matches, will the first or all of the increments be added to the baseline timeout?

File: test/spec/modules/timeoutRtdProvider_spec.js
Comment (patmmccann) [inline test/spec/modules/timeoutRtdProvider_spec.js:296]: can you add a unit test for the multiple rules match case?

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/timeoutRtdProvider.js
Comment (patmmccann) [inline modules/timeoutRtdProvider.js:31]: might be useful to consider outstream multiformat units differently?

**Review by patmmccann — CHANGES_REQUESTED** [review]
Mostly documentation changes, one additional unit test

File: modules/timeoutRtdProvider.md
Comment (patmmccann) [reply modules/timeoutRtdProvider.md:15]: actually you do say this, but perhaps reorder the sentence a small bit

`The timeout RTD module enables publishers to set rules  that determine the timeout based 
on certain features. It supports rule sets dynamically retrieved from a timeout provider as well as rules set directly via configuration`

File: test/spec/modules/timeoutRtdProvider_spec.js
Comment (luigi-sayson) [reply test/spec/modules/timeoutRtdProvider_spec.js:296]: This should be handled by this test case: https://github.com/prebid/Prebid.js/pull/7395/files#diff-49900727474e17b1862be0b57a305580c07d69ca22e2e4c924a78d0630631735R173-R205, though I added one anyway, since this new one also checks if the timeout is incremented properly when there's multiple matching rules

File: modules/timeoutRtdProvider.js
Comment (luigi-sayson) [reply modules/timeoutRtdProvider.js:31]: Yeah, though I think we'll just save that for a future version of this module

**Comment by patmmccann** [comment]
LGTM ; please make a docs repo PR before we merge

**Review by patmmccann — APPROVED** [review]

**Comment by luigi-sayson** [comment]
👍 
Done: https://github.com/prebid/prebid.github.io/pull/3266

**Comment by patmmccann** [comment]
@ChrisHuie ; as this is a general module, I should have looked for second approval. Could you take a look as well?

**Pull request merged**
